data/json/items/tool/electronics.json

```diff
@@ -130,6 +130,24 @@
       }
     ]
   },
+  {
+    "id": "scenario_cell_phone",
```

Comment:
Is it not working with existing item?

Reply:
![Screenshot from 2020-07-05 10-35-57](https://user-images.githubusercontent.com/67556457/86535047-759b0480-beab-11ea-9f69-6a081d8c2098.png)


That is without my fix. It will spawn a cellphone with no charges. IF I change the charged_cell_phone group to spawn a charged battery, it will pull the battery charges correctly on spawning, but not the correct battery. The cell phone will spawn with a plutonium battery that has 150/1000 charges.

When checking the cell_phone in item.json, I removed the entries for plutonium battery and left behind

`"pocket_data": [
      {
        "pocket_type": "MAGAZINE_WELL",
        "rigid": true,
        "holster": true,
        "max_contains_volume": "20 L",
        "max_contains_weight": "20 kg",
        "item_restriction": [
          "light_plus_battery_cell",
          "light_minus_battery_cell",
          "light_minus_disposable_cell",
          "light_battery_cell",
          "light_disposable_cell",
        ]`

Then it spawned with a battery charge of 100. So I was a little confused about why that happened. It made me think that somehow the last few items in the list were being considered instead of selecting a matching battery with charges.

So then I did:

`"pocket_data": [
      {
        "pocket_type": "MAGAZINE_WELL",
        "rigid": true,
        "holster": true,
        "max_contains_volume": "20 L",
        "max_contains_weight": "20 kg",
        "item_restriction": [
          "light_minus_battery_cell",
          "light_minus_disposable_cell",
          "light_battery_cell",
          "light_disposable_cell",
          "light_atomic_battery_cell",
          "light_minus_atomic_battery_cell"
          "light_plus_battery_cell",
        ]
      }
    ]
  },`

And that did not change the situation.

I think somewhere along the line there may be an issue with how information is pulled from magazine well in pocket data. It wants to take the maximum value of whatever battery has the highest charges and implant that into the phone. That's how users are ending up with either no charges or they're ending up with a plutonium battery in their phones.

Or I could just be wrong about my assessment.

Reply:
OR I could be building the item group incorrectly. But I tried to model it based off of smart_phone and that is what ultimately got me the correct results.

Reply:
I also did not want to touch the original as it is possible it is modified elsewhere within mapgens or other itemgroups.

Reply:
No.

Reply:
The issue has been tracked to #41856 I think.

I just figured out that it is loading the magazine based on alphabetical order.